src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,381 @@
+    }
+    svgRanges.reverse();
+    return (
+      <div
+        class="svgContainer"
+        style={{
+          height: `${this.minorSize * 1}px`,
+          width: `${this.minorSize * 1}px`
+        }}
+      >
+        <svg width="100%" height="100%" viewBox="0 0 100 100">
+          {svgRanges}
+        </svg>
+        <div
+          class="gaugeContainer"
+          style={{
+            "box-shadow":
+              (this.minorSize <= 300 && this.thickness <= 25) ||
+              !this.styleShadow
+                ? "none"
+                : "",
+            height: `${this.minorSize * 0.8 +
+              (this.calcThickness() * (this.minorSize / 100) -
+                this.minorSize / 100)}px`,
+            width: `${this.minorSize * 0.8 +
+              (this.calcThickness() * (this.minorSize / 100) -
+                this.minorSize / 100)}px`
+          }}
+        />
+        {this.showValue ? (
+          <span
+            class="marker"
+            style={{
+              display: this.showValue ? "" : "none",
+              height: `${this.minorSize * 0.795 -
+                this.calcThickness() * (this.minorSize / 100)}px`,
+              transform:
+                this.calcPercentage() >= 100
+                  ? "rotate(359deg)"
+                  : this.calcPercentage() > 0
+                    ? `rotate(${3.6 * this.calcPercentage()}deg)`
+                    : "rotate(0.5deg)"
+            }}
+          >
+            <div
+              class="indicator"
+              style={{
+                "box-shadow":
+                  (this.minorSize <= 300 && this.thickness <= 25) ||
+                  !this.styleShadow
+                    ? "none"
+                    : "",
+                height:
+                  this.minorSize / 100 +
+                  this.calcThickness() * (this.minorSize / 150) +
+                  "px",
+                transform:
+                  "translateY(-" +
+                  (this.minorSize / 100 +
+                    this.calcThickness() * (this.minorSize / 150)) +
+                  "px)"
+              }}
+            />
+          </span>
+        ) : (
+          ""
+        )}
+        <div
+          class="gauge"
+          style={{
+            "box-shadow":
+              (this.minorSize <= 300 && this.thickness <= 25) ||
+              !this.styleShadow
+                ? "none"
+                : "",
+            height: `${this.minorSize * 0.792 -
+              this.calcThickness() * (this.minorSize / 100)}px`,
+            width: `${this.minorSize * 0.792 -
+              this.calcThickness() * (this.minorSize / 100)}px`
+          }}
+        >
+          {this.showValue ? (
+            <div
+              style={{
+                "font-size": `${(this.minorSize * 0.795 -
+                  this.calcThickness() / 2 * (this.minorSize / 100)) /
+                  8}px`
+              }}
+            >
+              {this.calcPercentage() > 100
```

Comment:
Consider this expression instead:

```suggestion
              {Math.min(Math.max(Math.round(this.calcPercentage()), 0), 100)}
```

Reply:
🤔 😲